client/src/components/QuestionsAnswers/QuestionForm.jsx

```diff
@@ -0,0 +1,158 @@
+    axios
+      .post('https://app-hrsei-api.herokuapp.com/api/fec2/hratx/qa/questions',
+        {
+          'product_id': productId,
+          'body': question,
+          'name': nickname,
+          'email': email
+        },
+        {
+          headers: {
+            'Authorization': API_KEY
+          }
+        })
+      .then(response => {
+        const newQuestion = [{
+          'question_id': null,
+          'question_body': question,
+          'question_date': null,
+          'asker_name': nickname,
+          'question_helpfulness': 0,
+          'reported': false,
+          'answers': {},
+        }];
+        handleNewQuestion(newQuestion);
+      })
+      .catch(err => {
+        console.log('Error: cannot add question');
+      });
+  };
+
+  const handleShow = () => {
+    setShow(true);
+  };
+  const handleClose = () => {
+    if (question.concat(nickname, email) === '') {
+      setShow(false);
+    } else if (confirm('Are you sure you want to exit? Your changes will not be saved.')) {
+      setShow(false);
+    }
+  };
+  const handleQuestionChange = (e) => {
+    setQuestion(e.target.value);
+  };
+  const handleNicknameChange = (e) => {
+    setNickname(e.target.value);
+  };
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+  const handleSubmit = (e) => {
+    const form = e.currentTarget;
+    if (form.checkValidity() === false) {
+      e.preventDefault();
+      e.stopPropagation();
+    } else {
+      e.preventDefault();
+      setShow(false);
+      submitQuestion();
+    }
+    setValidated(true);
+  };
+
+  useEffect(() => {
+    getProductName();
+  }, []);
+
+  return (
+    <div>
+      {(!length) ? <Button onClick={handleShow}>Submit a New Question</Button> : <Button onClick={handleShow}>Add a Question + </Button>}
+      <Modal show={show} onHide={handleClose} backdrop='static' keyboard={false} centered>
+        <Modal.Header closeButton>
+          <Modal.Title>
+            Ask Your Question<br></br>
+            <small><em>About the {product}</em></small>
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p><em>All fields marked with a * are mandatory.</em></p>
+          <Form noValidate validated={validated} onSubmit={handleSubmit}>
+            <Form.Group controlId='question'>
+              <Form.Label>What is your question?*</Form.Label>
+              <InputGroup hasValidation>
+                <Form.Control
+                  as='textarea'
+                  placeholder='Write your question here'
+                  maxLength='1000'
+                  rows={3}
+                  onChange={handleQuestionChange}
+                  required />
+                <Form.Control.Feedback type='invalid'>Please enter a question.</Form.Control.Feedback>
+              </InputGroup>
+            </Form.Group>
+            <br />
+            <Form.Group controlId='nickname'>
+              <Form.Label>What is your nickname?*</Form.Label>
```

Comment:
Am I reading this correctly that the user has to give a nickname?

Reply:
yup, the nickname is a mandatory input field for the Question Form!

Reply:
How strange.